File: 2 - Coin Flipping RNG Chance Calculator/random.java
import java.lang.Math;

public class MyProgram
{
    public static void main(String[] args)
    {
        PRNG prng = new PRNG();
        prng.flip(9999999);
        
    }
}

class PRNG 
{
    
    
    public PRNG() {
        
    }
    public void flip(int flips) {
        
        int heads = 0;
        int tails = 0;
        for (int x = 0; x < flips; x++) {
            int randNum = (int) Math.round(Math.random());
            if (randNum == 0) {
                heads++;
            }
            else if (randNum == 1) {
                tails++;
            }
            else {
                System.out.println("Error: Invalid Number Recieved");
                break;
                
            }
        }
        System.out.println("Heads/flips: " + ((double)heads/(double)flips)*100 + "%");
        System.out.println("Tails/flips: " + ((double)tails/(double)flips)*100 + "%");
        System.out.println("Heads Total: " + heads);
        System.out.println("Tails Total: " + tails);
        System.out.println("Number of Flips: " + flips);
        
    }
}

File: 2 - Coin Flipping RNG Chance Calculator/random.cpp
#include <iostream>
#include <cmath>
using namespace std;
/*
 * Generates a Random Number being either 0 or 1
 * 0 = Heads
 * 1 = Tails
 **************************************************
 *
 * Author: Aaron Buckner
 * License: MIT
 * Start Date: 9/7/23 @ 12:55 PM CST
 * Last Edited:  9/12/23 @ 05:12 PM CST
 *
 */
class coinFlip {
  private:
  public:
    int heads;
    int tails;
    int flips;
    void flip(int flips) {
        srand(time(0));
        
        for (int x = 0; x < flips; x++) {
            srand(rand()%(2147483647));
            int randNum = rand()%(2);
            if (randNum == 0) {
                heads++;
            }
            else if (randNum == 1) {
                tails++;
            }
            else {
                throw;
            }
        }
        cout << "Heads/flips: "<< ((double)heads/(double)flips)*100 << "%" << endl;
        cout << "Tails/flips: "<< ((double)tails/(double)flips)*100 << "%" << endl;
        cout << "Heads Total: " << heads << endl;
        cout << "Tails Total: " << tails << endl;
        cout << "Number of Flips: " << flips << endl;
         
    }
};

int main()
{
    
    cout << "Flipping has begun, please be patient." << endl;
    
    int heads = 0;
    int tails = 0;
    int flips = 9999999; //9,999,999 flips
    
    coinFlip flipper;
    flipper.flip(flips);
    
    
}
